analyzer/runtime/extract.go

```diff
@@ -448,7 +448,9 @@ func ExtractRound(blockHeader nodeapi.RuntimeBlockHeader, txrs []nodeapi.Runtime
 					}
 
 					// Handle encrypted txs.
-					if evmEncrypted, err2 := evm.EVMMaybeUnmarshalEncryptedData(body.InitCode, ok); err2 == nil {
+					// We don't pass the tx result to EVMMaybeUnmarshalEncryptedData because it's the
+					// address of the created contract, which is already tracked in blockTransactionData.To.
```

Comment:
```suggestion
					// We don't pass the tx result (`ok`) to EVMMaybeUnmarshalEncryptedData because it's the 
					// (unencrypted) address of the created contract. The function expects a CBOR-encoded 
					// encryption envelope as its second argument, so passing the unencrypted address
					// makes it incorrectly declare the whole tx unencrypted.
					// Note: The address of the new contract is tracked in blockTransactionData.To.
```

^ To explain why we _have_ to omit `ok`; your comment only explains why we _can_.

Reply:
oop that's better thanks